mrsal/mrsal.py

```diff
@@ -322,33 +322,37 @@ def start_consumer(
         self.log.info(f'Consuming messages: queue= {queue}, requeue= {requeue}, inactivity_timeout= {inactivity_timeout}')
 
         try:
+            method_frame: spec.Basic.Deliver 
+            prop: spec.BasicProperties 
+            body: Any
             for method_frame, properties, body in \
                     self._channel.consume(queue=queue, inactivity_timeout=inactivity_timeout):
                 try:
                     if (method_frame, properties, body) != (None, None, None):
                         consumer_tags = self._channel.consumer_tags
-                        consumer_tag = method_frame.consumer_tag
+                        self.consumer_tag = method_frame.consumer_tag
+                        app_id = properties.app_id
+                        msg_id = properties.message_id
```

Comment:
I don't understand properly what the benefits of having this in the wrapper is. Are they optional?

Reply:
- First of all app_id and msg_id are optional. 
- These values can be set by the publisher when we publish the msg to an exch.
- After that, we can read these values in the consumer when we consume the msg.
-  > Here I'm just reading these values in the consumer and logging them.

- The new thing is that the **callback** method that we pass to the consumer to process the consumed message sometimes needs extra info about the message (e.g: 
   - Msg type
   - messageId specified by the publisher (It can be the name of the file sent by the NeoGate publisher)
   - app_id specified also by the publisher (It can be NeoGate, DICOM_SERVER or the name of the app that published to the exchange)

But I need to reformat the code to pass these values to the **callback** method just if **explicitly** specified by the user.